File: Java/Problems/src/main/java/LeetCode/ContainsDuplicate.java
package LeetCode;

import java.util.HashMap;

//https://leetcode.com/problems/contains-duplicate/
public class ContainsDuplicate {
    public Boolean containsDuplicate(int[] nums) {
        HashMap<Integer, Integer> map = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int value = nums[i];
            if (map.containsKey(value)) {
                return true;
            }
            map.put(value, i);
        }

        //don't found duplicates.
        return false;
    }
}

File: Java/Problems/src/main/kotlin/ContainsDuplicate.kt
class ContainsDuplicate {
    //https://leetcode.com/problems/contains-duplicate/
    fun containsDuplicate(nums: IntArray): Boolean {
        val map = mutableMapOf<Int, Int>()

        for (i in nums.indices){
            if (map.containsKey(i)) return true

            map[nums[i]] = i
        }
        return false
    }
}